File: cases/core/synthetic_vulnerable/PY-SV-002/pr/base/agent/router.py
"""Agent router — dispatches plan steps to the appropriate tool."""

from __future__ import annotations

import logging

from agent.planner import PlanStep

logger = logging.getLogger(__name__)


def execute_plan(plan: list[PlanStep]) -> list[dict]:
    """Execute each step in the export plan and collect results."""
    results: list[dict] = []

    for step in plan:
        if not isinstance(step, dict):
            logger.warning("Skipping non-dict plan step: %r", step)
            continue

        action = step.get("action")
        if not isinstance(action, str):
            results.append({
                "path": "",
                "status": "error",
                "reason": "Plan step missing valid 'action' key",
            })
            continue

        results.append({
            "path": step.get("output_path", ""),
            "status": "skipped",
            "reason": f"Unknown action: {action}",
        })

    return results

File: cases/core/capability_safe/TS-CS-001/pr/base/agent/router.ts
/** Agent router — dispatches plan steps to the appropriate tool. */

import { PlanStep } from "./planner";

export interface ApiResult {
  url: string;
  status: string;
  summary: string;
}

export async function executePlan(plan: PlanStep[]): Promise<ApiResult[]> {
  const results: ApiResult[] = [];

  for (const step of plan) {
    results.push({
      url: step.url ?? "",
      status: "skipped",
      summary: `No tool registered for action: ${step.action}`,
    });
  }

  return results;
}
